src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.equals(ProductServiceImpl.class)
+                && !interfaceClazz.equals(ProductService.class)
+                && !interfaceClazz.equals(ProductParser.class)
+                && !interfaceClazz.equals(ProductParserImpl.class)
+                && !interfaceClazz.equals(FileReaderServiceImpl.class)
+                && !interfaceClazz.equals(FileReaderService.class)) {
+            throw new RuntimeException("Unsupported class:" + interfaceClazz);
+        }
+        Object clazzImplementationInstance = null;
+        if (interfaceClazz.equals(ProductService.class)) {
+            interfaceClazz = ProductServiceImpl.class;
+        }
```

Comment:
You didnt check if class has Component annotation in your Injector.
You should check this. And if class didnt have a Component annotation you shouldn't pass.

Reply:
I think Component is just for reference. We check fields annotations only. I mean Inject. If I delete @componet form here http://prntscr.com/1thpf6o. Code works.

Reply:
Task said us "If Component annotation is missing above the class, we shouldn't be able to create an instance of this class in Injector, and we should throw an exception"